glucose build fails on musllinux

```
 [glucose-4.2.1]   [spkg-install] In file included from /host/sage-musllinux_1_2_x86_64/var/tmp/sage/build/glucose-4.2.1/src/simp/../core/Solver.cc:52:
  [glucose-4.2.1]   [spkg-install] /host/sage-musllinux_1_2_x86_64/var/tmp/sage/build/glucose-4.2.1/src/simp/../utils/System.h:25:10: fatal error: fpu_control.h: No such file or directory
  [glucose-4.2.1]   [spkg-install]    25 | #include <fpu_control.h>
  [glucose-4.2.1]   [spkg-install]       |          ^~~~~~~~~~~~~~~
  [glucose-4.2.1]   [spkg-install] compilation terminated.
  [glucose-4.2.1]   [spkg-install] make[3]: *** [/host/sage-musllinux_1_2_x86_64/var/tmp/sage/build/glucose-4.2.1/src/simp/../mtl/template.mk:71: /host/sage-musllinux_1_2_x86_64/var/tmp/sage/build/glucose-4.2.1/src/simp/../core/Solver.o] Error 1
```
https://github.com/mkoeppe/sage/actions/runs/15521576042/job/43700504406#step:12:40545

--- a/build/pkgs/glucose/patches/0002-fix-FPU-control-preventing-compilation-on-ARM-CPUs.patch
+++ b/build/pkgs/glucose/patches/0002-fix-FPU-control-preventing-compilation-on-ARM-CPUs.patch
@@ -1,7 +1,7 @@
 From c9f3ec32c92971bd193147a98a956b6876caf1be Mon Sep 17 00:00:00 2001
 From: Petr Ambroz <petrambroz@icloud.com>
 Date: Tue, 5 Nov 2024 12:29:01 +0000
-Subject: [PATCH] fix FPU control preventing compilation on ARM CPUs
+Subject: [PATCH 2/3] fix FPU control preventing compilation on ARM CPUs
 
 ---
  parallel/Main.cc | 2 +-
@@ -34,3 +34,6 @@ index c2327df..8302179 100644
          fpu_control_t oldcw, newcw;
          _FPU_GETCW(oldcw); newcw = (oldcw & ~_FPU_EXTENDED) | _FPU_DOUBLE; _FPU_SETCW(newcw);
          //printf("c WARNING: for repeatability, setting FPU to use double precision\n");
+-- 
+2.47.0
+

--- a/build/pkgs/glucose/patches/0003-Only-use-fpu_control.h-with-__GLIBC__.patch
+++ b/build/pkgs/glucose/patches/0003-Only-use-fpu_control.h-with-__GLIBC__.patch
@@ -1,0 +1,53 @@
+From 2e49aeadc216062efd5cfd496c68610ee2a2eda2 Mon Sep 17 00:00:00 2001
+From: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
+Date: Mon, 9 Jun 2025 17:27:23 -0700
+Subject: [PATCH 3/3] Only use fpu_control.h with __GLIBC__
+
+---
+ parallel/Main.cc | 2 +-
+ simp/Main.cc     | 2 +-
+ utils/System.h   | 2 +-
+ 3 files changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/parallel/Main.cc b/parallel/Main.cc
+index 00d044f..423c961 100644
+--- a/parallel/Main.cc
++++ b/parallel/Main.cc
+@@ -97,7 +97,7 @@ int main(int argc, char** argv)
+         setUsageHelp("c USAGE: %s [options] <input-file> <result-output-file>\n\n  where input may be either in plain or gzipped DIMACS.\n");
+         // printf("This is MiniSat 2.0 beta\n");
+         
+-#if defined(__linux__) && (defined(__i386__) || defined(__x86_64__))
++#if defined(__linux__) && defined(__GLIBC__) && (defined(__i386__) || defined(__x86_64__))
+         fpu_control_t oldcw, newcw;
+         _FPU_GETCW(oldcw); newcw = (oldcw & ~_FPU_EXTENDED) | _FPU_DOUBLE; _FPU_SETCW(newcw);
+         printf("c WARNING: for repeatability, setting FPU to use double precision\n");
+diff --git a/simp/Main.cc b/simp/Main.cc
+index 8302179..336a7aa 100644
+--- a/simp/Main.cc
++++ b/simp/Main.cc
+@@ -122,7 +122,7 @@ int main(int argc, char** argv)
+       setUsageHelp("c USAGE: %s [options] <input-file> <result-output-file>\n\n  where input may be either in plain or gzipped DIMACS.\n");
+ 
+ 
+-#if defined(__linux__) && (defined(__i386__) || defined(__x86_64__))
++#if defined(__linux__) && defined(__GLIBC__) && (defined(__i386__) || defined(__x86_64__))
+         fpu_control_t oldcw, newcw;
+         _FPU_GETCW(oldcw); newcw = (oldcw & ~_FPU_EXTENDED) | _FPU_DOUBLE; _FPU_SETCW(newcw);
+         //printf("c WARNING: for repeatability, setting FPU to use double precision\n");
+diff --git a/utils/System.h b/utils/System.h
+index 4e48fee..7fb9e81 100644
+--- a/utils/System.h
++++ b/utils/System.h
+@@ -21,7 +21,7 @@ OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWA
+ #ifndef Glucose_System_h
+ #define Glucose_System_h
+ 
+-#if defined(__linux__)
++#if defined(__linux__) && defined(__GLIBC__) && (defined(__i386__) || defined(__x86_64__))
+ #include <fpu_control.h>
+ #endif
+ 
+-- 
+2.47.0
+

